src/cpp/qos/ReaderQos.cpp

```diff
@@ -26,94 +26,89 @@ namespace fastrtps {
-		m_liveliness.hasChanged = true;
-	}
-	if(first_time)
-	{
-		m_reliability = qos.m_reliability;
-		m_reliability.hasChanged = true;
-	}
-	if(first_time)
-	{
-		m_ownership = qos.m_ownership;
-		m_ownership.hasChanged = true;
-	}
-	if(m_destinationOrder.kind != qos.m_destinationOrder.kind)
-	{
-		m_destinationOrder = qos.m_destinationOrder;
-		m_destinationOrder.hasChanged = true;
-	}
-	if (m_userData.getDataVec() != qos.m_userData.getDataVec())
-	{
-		m_userData = qos.m_userData;
-		m_userData.hasChanged = true;
-	}
-	if(m_timeBasedFilter.minimum_separation != qos.m_timeBasedFilter.minimum_separation )
-	{
-		m_timeBasedFilter = qos.m_timeBasedFilter;
-		m_timeBasedFilter.hasChanged = true;
-	}
-	if(first_time || m_presentation.access_scope != qos.m_presentation.access_scope ||
-			m_presentation.coherent_access != qos.m_presentation.coherent_access ||
-			m_presentation.ordered_access != qos.m_presentation.ordered_access)
-	{
-		m_presentation = qos.m_presentation;
-		m_presentation.hasChanged = true;
-	}
-	if(qos.m_partition.getNames().size()>0)
-	{
-		m_partition = qos.m_partition;
-		m_partition.hasChanged = true;
-	}
-	if (m_topicData.getValue() != qos.m_topicData.getValue())
-	{
-		m_topicData = qos.m_topicData;
-		m_topicData.hasChanged = true;
-	}
-	if(m_groupData.getValue() != qos.m_groupData.getValue() )
-	{
-		m_groupData = qos.m_groupData;
-		m_groupData.hasChanged = true;
-	}
-	if(first_time || m_durabilityService.history_kind != qos.m_durabilityService.history_kind ||
-			m_durabilityService.history_depth != qos.m_durabilityService.history_depth ||
-			m_durabilityService.max_instances != qos.m_durabilityService.max_instances ||
-			m_durabilityService.max_samples != qos.m_durabilityService.max_samples||
-			m_durabilityService.max_samples_per_instance != qos.m_durabilityService.max_samples_per_instance ||
-			m_durabilityService.service_cleanup_delay != qos.m_durabilityService.service_cleanup_delay
-	)
-	{
-		m_durabilityService = qos.m_durabilityService;
-		m_durabilityService.hasChanged = true;
-	}
-	if(m_lifespan.duration != qos.m_lifespan.duration )
-	{
-		m_lifespan = qos.m_lifespan;
-		m_lifespan.hasChanged = true;
-	}
+    if(first_time)
+    {
+        m_durability = qos.m_durability;
+        m_durability.hasChanged = true;
+    }
+    if(first_time || m_deadline.period != qos.m_deadline.period)
+    {
+        m_deadline = qos.m_deadline;
+        m_deadline.hasChanged = true;
+    }
+    if(m_latencyBudget.duration != qos.m_latencyBudget.duration)
+    {
+        m_latencyBudget = qos.m_latencyBudget;
+        m_latencyBudget.hasChanged = true;
+    }
+    if (first_time)
+    {
+        m_liveliness = qos.m_liveliness;
+        m_liveliness.hasChanged = true;
+    }
+    if(first_time)
+    {
+        m_reliability = qos.m_reliability;
+        m_reliability.hasChanged = true;
+    }
+    if(first_time)
+    {
+        m_ownership = qos.m_ownership;
+        m_ownership.hasChanged = true;
+    }
```

Comment:
The above three `if`s could be combined, but I assume as this is a backport we shouldn't aim at making any improvements?

Reply:
I doubt about this kind of things. I saw other parts of the code that may be improved, but as you said, as this is a backport I don't think we should waste time in this kind of things.

Reply:
Yeap, I agree, we should leave it as it is.